### PR TITLE
Pv2 1030 pv2 1031 pv2 695 pv2 694 approvals handle pause and resume

### PR DESCRIPTION
**Pull request by samuel-dev-dba**

### CODE REVIEW

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs [hunk 1]
```diff
@@ -410,7 +425,111 @@ public void WhenTheApprovalsServiceNotifiesThePaymentsServiceThatTheApprenticesh
...
+                return false;
+            }, "Failed to find all the stored apprenticeships paused details.");
+        }
+
+        [When(@"the Approvals service notifies the Payments service that the apprenticeships has been resumed")]
+        public void WhenTheApprovalsServiceNotifiesThePaymentsServiceThatTheApprenticeshipsHasBeenResumed()
+        {
+            foreach (var approvalsApprenticeship in ApprovalsApprenticeships)
+            {
+                var (employer, sendingEmployer, provider, learner) = GetApprovalsReferenceData(approvalsApprenticeship);
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs]: not used. Can it be removed?

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs [hunk 2]
```diff
@@ -425,16 +544,18 @@ private ApprenticeshipModel CreateApprenticeshipModel(ApprovalsApprenticeship ap
...
                 LegalEntityName = employer.Name,
                 EstimatedStartDate = apprenticeshipSpec.StartDate.ToDate(),
                 EstimatedEndDate = apprenticeshipSpec.EndDate.ToDate(),
                 AgreedOnDate = string.IsNullOrWhiteSpace(apprenticeshipSpec.AgreedOnDate)
                     ? DateTime.UtcNow
                     : apprenticeshipSpec.AgreedOnDate.ToDate(),
                 IsLevyPayer = true,
-                StopDate = string.IsNullOrWhiteSpace(apprenticeshipSpec.StoppedOnDate) 
-                    ? default(DateTime?) 
+                StopDate = string.IsNullOrWhiteSpace(apprenticeshipSpec.StoppedOnDate)
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs]: use extension method

File: src/SFA.DAS.Payments.DataLocks.Domain.UnitTests/ApprenticeshipPausedServiceTest.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+using Autofac.Extras.Moq;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.Payments.DataLocks.Domain.Models;
+using SFA.DAS.Payments.DataLocks.Domain.Services.Apprenticeships;
+using SFA.DAS.Payments.Model.Core.Entities;
+
+namespace SFA.DAS.Payments.DataLocks.Domain.UnitTests.Services
+{
+    public class ApprenticeshipPausedServiceTest
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.DataLocks.Domain.UnitTests/ApprenticeshipPausedServiceTest.cs]: this is a duplicate class definition. The same class is in the Services folder. Causes build to fail (assume it's a drag and drop issue)

File: src/SFA.DAS.Payments.DataLocks.Domain/Services/Apprenticeships/ApprenticeshipUpdatedService.cs [hunk 4]
```diff
@@ -43,6 +45,11 @@ public async Task<ApprenticeshipModel> UpdateApprenticeship(T updatedApprentices
 
         protected abstract void HandleUpdated(ApprenticeshipModel current, T updated);
 
+        protected virtual Task UpdateHistoryTables(T updated)
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.DataLocks.Domain/Services/Apprenticeships/ApprenticeshipUpdatedService.cs]: is this going to be used later?

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs [hunk 5]
```diff
@@ -410,7 +425,111 @@ public void WhenTheApprovalsServiceNotifiesThePaymentsServiceThatTheApprenticesh
...
+                var apprenticeshipIds = ApprovalsApprenticeships.Select(apprenticeship => apprenticeship.Id).ToArray();
+
+                var savedApprenticeshipPauses = await TestDataContext.ApprenticeshipPause.AsNoTracking()
+                    .Where(o => apprenticeshipIds.Contains(o.ApprenticeshipId))
+                    .ToListAsync();
+
+                var notFound = new List<ApprovalsApprenticeship>();
+                foreach (var approvalsApprenticeship in ApprovalsApprenticeships)
+                {
+                    var (employer, sendingEmployer, provider, learner) = GetApprovalsReferenceData(approvalsApprenticeship);
```
[the remark this answers is not in the document]
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs]: The learner property is used to get the Uln

File: src/SFA.DAS.Payments.DataLocks.Domain/Services/Apprenticeships/ApprenticeshipUpdatedService.cs (hunk 4 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.DataLocks.Domain/Services/Apprenticeships/ApprenticeshipUpdatedService.cs]: Yes is used in the Derived class Pause Service

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs [hunk 6]
```diff
@@ -410,7 +425,111 @@ public void WhenTheApprovalsServiceNotifiesThePaymentsServiceThatTheApprenticesh
...
             }
         }
-        
+
+        [When(@"the Approvals service notifies the Payments service that the apprenticeships has been paused")]
+        public void WhenTheApprovalsServiceNotifiesThePaymentsServiceThatTheApprenticeshipsHasBeenPaused()
+        {
+            foreach (var approvalsApprenticeship in ApprovalsApprenticeships)
+            {
+                var (employer, sendingEmployer, provider, learner) = GetApprovalsReferenceData(approvalsApprenticeship);
```
[the remark this answers is not in the document]
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs]: Removed

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs (hunk 1 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs]: Removed

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs (hunk 2 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs]: done

File: src/SFA.DAS.Payments.DataLocks.Domain.UnitTests/ApprenticeshipPausedServiceTest.cs (hunk 3 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.DataLocks.Domain.UnitTests/ApprenticeshipPausedServiceTest.cs]: Removed

**Pull request merged**